src/gtest/test_joinsplit.cpp

```diff
@@ -579,7 +579,7 @@ TEST(joinsplit, note_plaintexts)
 
     // Check memo() returns by reference, not return by value, for use cases such as:
     // std::string data(plaintext.memo().begin(), plaintext.memo().end());
-    ASSERT_TRUE(decrypted.memo().data() == decrypted.memo().data());
+    ASSERT_TRUE(decrypted.memo().data() == decrypted.memo().data()); //-V501

```

Comment:
This assert is strange, I think it shouldbe:
ASSERT_TRUE(decrypted.memo().data() == note_pt.memo().data());

Reply:
ASSERT_TRUE(decrypted.memo().data() == note_pt.memo().data());
It is false.
data() is a pointer.
so it is different.

Reply:
So it was checked before.
ASSERT_TRUE(decrypted.memo() == note_pt.memo());

Reply:
I see. That was my bad for not checking what data() returns.

Reply:
They were checking here that decrypted.memo() is a reference.
There is a better way to check if it memo() points to reference using c++17:
ASSERT_TRUE(std::is_lvalue_reference<decltype(decrypted.memo())>::value);

So, decltype - we are getting data type from decrypted.memo(), then checking if it is lvalue reference.

Reply:
Nice, cool trick